R/doeFactorial.R

```diff
@@ -175,100 +181,200 @@ doeFactorial <- function(jaspResults, dataset, options, ...) {
            rnd <- FALSE,
            rnd <- TRUE)
 
-    table$addColumnInfo(name = 'runOrder', title = gettext("RunOrder"), type = 'string')
+    ifelse(options[["factorialRepeats"]],
+           rep <- TRUE,
+           rep <- FALSE)
 
+    table$addColumnInfo(name = 'runOrder', title = gettext("Run order"), type = 'string')
+
+    #default generators
     if (options[["factorialType"]] == "factorialTypeDefault") {
       if (options[["designBy"]] == "designByRuns") {
```

Comment:
There are repeating `if` statements like `if (options[["factorialType"]] == "factorialTypeDefault")` and `if (options[["factorialType"]] == "factorialTypeSplit") {`. Since these can never both be true, all if statements after the first should be `else if`.